client/components/selected-image.jsx

```diff
@@ -1,20 +1,81 @@
 import React, { useState, useEffect } from 'react';
 
-export default function SelectedImage({ imageId }) {
+export default function SelectedImage({ imageId, user }) {
   const [image, setImage] = useState('');
+  const [isLiked, setIsLiked] = useState(false);
 
+  // Fetch the selected image's row from the database
   useEffect(() => {
     fetch(`/images/${imageId}`)
       .then(response => response.json())
-      .then(data => setImage(data));
+      .then(data => setImage(data))
+      .catch(error => console.error(error));
   }, [imageId]);
 
+  // Fetch a list of all liked images and check to see if the
+  // selected image is liked by the current user
+  useEffect(() => {
+    fetch(`/images/${imageId}/likedImage`)
+      .then(response => response.json())
+      .then(data => {
+        for (let i = 0; i < data.length; i++) {
+          const currUser = data[i].userId;
+          const currImg = data[i].imageId;
+          if (user && currUser === String(user.userId) && currImg === imageId) {
+            setIsLiked(true);
+            return;
+          } else {
+            setIsLiked(false);
+          }
+        }
+      })
+      .catch(error => console.error(error));
+  }, [imageId, user, isLiked]);
+
+  // Adds the current user's userId and the selected imageId to the database
```

Comment:
I'm loving these comments! They make it so much easier for me to know what I'm looking at when I read the code.